### PR TITLE
tests: More cleanup to tests using profiles

### PR DESCRIPTION
**Pull request by spencer-lunarg**
The following tests were not working with local max profile

### CODE REVIEW

**Review by juan-lunarg — APPROVED** [review]
:+1:

File: tests/vklayertests_others.cpp [hunk 1]
```diff
@@ -4587,18 +4587,13 @@ TEST_F(VkLayerTest, QueryPerformanceCreation) {
...
     auto performance_features = LvlInitStruct<VkPhysicalDevicePerformanceQueryFeaturesKHR>();
-    features2 = LvlInitStruct<VkPhysicalDeviceFeatures2KHR>(&performance_features);
-    vkGetPhysicalDeviceFeatures2KHR(gpu(), &features2);
+    auto features2 = GetPhysicalDeviceFeatures2(performance_features);
     if (!performance_features.performanceCounterQueryPools) {
         GTEST_SKIP() << "Performance query pools are not supported.";
     }
 
-    ASSERT_NO_FATAL_FAILURE(InitState(nullptr, &performance_features));
+    ASSERT_NO_FATAL_FAILURE(InitState(nullptr, &features2));
```
Comment (ncesario-lunarg) [inline tests/vklayertests_others.cpp:4596]: Are `features2` necessary, or are only the features we care about present in `performance_features`? Similar comments for below instances.

**Review by ncesario-lunarg — APPROVED** [review]
One question, but LGTM!

File: tests/vklayertests_others.cpp (hunk 1 above)
Comment (spencer-lunarg) [reply tests/vklayertests_others.cpp:4596]: nope you are right, changing back, good catch

**Pull request merged**